services/worker-manager/test/api_test.js

```diff
@@ -1125,6 +1138,178 @@ helper.secrets.mockSuite(testing.suiteName(), ['db'], function(mock, skipping) {
       assert(scopes.has(`assume:worker-id:${workerGroup}/${awsWorkerIdentityProofParsed.instanceId}`), msg);
       assert(scopes.has(`secrets:get:worker-pool:${workerPoolId}`), msg);
       assert(scopes.has(`queue:claim-work:${workerPoolId}`), msg);
+      assert(scopes.has(`worker-manager:reregister-worker:${workerPoolId}/${workerGroup}/${awsWorkerIdentityProofParsed.instanceId}`), msg);
+    });
+  });
+
+  suite('reregisterWorker', function() {
+    const providerId = 'testing1';
+    const workerGroup = 'wg';
+    const workerId = 'wi';
+    const workerIdentityProof = {'token': 'tok'};
+
+    suiteSetup(function() {
+      helper.load.save();
+
+      // create fake clientId / accessToken for temporary creds
+      helper.load.cfg('taskcluster.credentials.clientId', 'fake');
+      helper.load.cfg('taskcluster.credentials.accessToken', 'fake');
+    });
+
+    suiteTeardown(function() {
+      helper.load.restore();
+    });
+
+    const defaultRegisterWorker = {
+      workerPoolId, providerId, workerGroup, workerId, workerIdentityProof,
+    };
+
+    test('works without registrationTimeout', async function() {
```

Comment:
These test cases look similar -- it might be helpful to factor them out a bit so that the reader sees only the things that differ from case to case.